src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlBinaryWriter.cs

```diff
@@ -625,18 +590,21 @@ public override unsafe void WriteText(string value)
         }
 
         public override unsafe void WriteText(char[] chars, int offset, int count)
+           => WriteText(chars.AsSpan(offset, count));
+
+        private unsafe void WriteText(Span<char> chars)
```

Comment:
```suggestion
        private unsafe void WriteText(ReadOnlySpan<char> chars)
```
I see your conversation above about that causing issues, but if it really resulted in things using the string version, as Jan says something is wrong that needs to be investigated.